index.js

```diff
@@ -173,4 +173,57 @@ mainForm.addEventListener('submit', (e) => {
     return false;
   }
   return true;
-});
\ No newline at end of file
+});
+
+// fetching enterd details from user
+const providedEmail = document.getElementById('email-input');
+const messageFromUser = document.getElementById('text-input');
+const dataByUser = {};
+const providedName = document.getElementById('name-input');
+
+// Local Data storage
+function isStorageSupported(type) {
+  let localStorage;
+  try {
+    localStorage = window[type];
+    const i = '__testing_storage__';
+    localStorage.setItem(i, i);
+    localStorage.removeItem(i);
+    return true;
+  } catch (e) {
+    return (
+      e instanceof DOMException
+      && (e.code === 22
+        || e.code === 1014
+        || e.name === 'QuotaExceededError'
+        || e.name === 'NS_ERROR_DOM_QUOTA_REACHED')
+        && localStorage
+        && localStorage.length !== 0);
+  }
+}
```

Comment:
- [ ] Good job implementing the local storage. I think it will be better if you implement the feature such that after filling the form, and not submitting it, the entered values should be saved in the browser 😄  and also be persistent on page reload. Please implement it 👍🏽

Reply:
Okay thanks very much